proposals/ref-readonly-parameters.md

```diff
@@ -159,7 +160,14 @@ Default parameter values could be an error for `ref readonly` parameters.
 
 Specifying the `RequiresLocationAttribute` in source could be allowed, similarly to `In` and `Out` attributes.
 
-Function pointer `ref readonly` parameters could be emitted without `modopt` or with `modreq`.
+Function pointer `ref readonly` parameters could be emitted with different `modopt`/`modreq` combinations:
+
+| Modifiers                             | Can be recognized across compilations | Old compilers see them as | `ref` → `ref readonly` | `in` → `ref readonly` |
+|---------------------------------------|---------------------------------------|---------------------------|------------------------|-----------------------|
+| `modreq(In) modopt(RequiresLocation)` | yes                                   | `in`                      | break                  | ok                    |
```

Comment:
I don't think there is a binary break - changing `modreq`/`modopt` seems to not have any effect on being able to call or pass around the function pointers in already-compiled code.

Reply:
Modopts and modreqs are reflected in the signature of the type. It is observable in this case when you change the return or parameter type of a method from `delegate*<ref int, void>` or `delegate*<in int, void>` to `delegate*<ref readonly int, void>`, as that will cause the parameter or return type to change.

Reply:
We can't really take binary breaks in the BCL. That includes adding or removing `modopt`/`modreq` or changing from one to the other.

That being said, I expect the number of people using `ref`/`in`/`out` in function pointers is fairly low and I don't believe we have any API surface in the BCL that would be impacted by how `ref readonly` plays with function pointers.  Notably with function pointers in particular you have an easy way to bypass things by simply casting from the actual signature to the desired signature or alternatively creating your own static wrapper method. This means users won't be "blocked" from making changes, they just may need to insert some new casts or helpers on recompile to keep things happy if they need to avoid any binary break.

I would expect this, overall, falls into the same camp as changing from `ref->in` on a virtual method. It's a break and people will want to avoid it. But, it won't be common and we have the tools (such as https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/overview) to help surface such binary breaks to the user. So simply emitting it as a `modreq`, much as we currently do for `in` and for virtual methods using such parameters, is fine.